### PR TITLE
fix: remove yarn, use npm

### PR DESCRIPTION
**Pull request by robertsLando**

### CODE REVIEW

**Comment by robertsLando** [comment]
@jmgiaever I have added some `FIXME` comment on places I'm not sure what to use

**Comment by jmgiaever** [comment]
Ty mate 🫶 Hah, I last night fixed stuff for 9.4.x 😆 I will try to transition for npm tomorrow. It's just burning around me atm. High season at work.

Why the change from yarn btw?

**Comment by robertsLando** [comment]
Hi Joachim 🫶 

I saw you pushed some stuff, feel free to edit the missing pieces here if needed (or tell me what to do, I don't know anything about snap), I would drop the `layout` part entirely but I'm not sure so I let you check it when you have time, just wanted to help with the transition to make it less painfull as I know you are very busy in this period :) 

> Why the change from yarn btw?

Long story short is it turned me crazy with bugs on docker, the only way I found to make everything work was to use some mixed npm commands, after making it work I just decided to drop yarn entirely as there was no reason to have both, now image size has been reduced a lot and it also take only 7 minutes to build on all archs compared to 1h 🎉

You can find the full story behind this here: https://github.com/zwave-js/zwave-js-ui/issues/3430#issuecomment-1825526240

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -127,52 +127,29 @@ parts:
...
-      chmod +x ./yarn
-      cd "${SNAPCRAFT_PART_BUILD}"
-
-      # Necessary for the remote build; sometimes.
-      if [ -n "${http_proxy-}" ]; then
-        yarn config set httpProxy "${http_proxy}"
-      fi
-      if [ -n "${https_proxy-}" ]; then
-        yarn config set httpsProxy "${https_proxy}"
-      fi
```
Comment (jmgiaever) [inline snap/snapcraft.yaml:144]: These should be there as it is a workaround for the remote build system. But they probably need to be changed for the npm ecosystem. 

But I can look into it tomorrow afternoon 👌 You probably need access to Snapcraft remote builds to debug this.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -95,7 +95,7 @@ apps:
   restart:
     command: bin/restart
 
-layout:
+layout: # FIXME: npm cache is on `~/.npm`
```
Comment (jmgiaever) [inline snap/snapcraft.yaml:98]: Means it's a subdirectory of the program files? I might have to do some tricks here, as that might be a Read Only partition.

Do you know if it's possible to specify the cache dir?

**Review by jmgiaever — COMMENTED** [review]
Note to myself: check prime on line 178, which excludes npm to be bundled with the snap. This is probably needed now. 

@robertsLando is npx needed?

File: snap/snapcraft.yaml (hunk 1 above)
Comment (jmgiaever) [reply snap/snapcraft.yaml:144]: @robertsLando If you happen to know the equivalent commands for npm, let me know. 🫶

Comment (robertsLando) [reply snap/snapcraft.yaml:144]: `npm config set` :)

**Comment by jmgiaever** [comment]
I might use them to parse the settings file back and fourth, in the Bourne scripts. Can't recall. I think it's time to rework the project as so much has changed the past years.

**Comment by jmgiaever** [comment]
> @robertsLando is npx needed?

**Comment by robertsLando** [comment]
Never used, dunno what is that for?

**Pull request merged**